lms/djangoapps/courseware/courses.py

```diff
@@ -152,10 +152,10 @@ def check_course_access(course, user, action, check_if_enrolled=False, check_sur
         # leaking info about access control settings
         raise CoursewareAccessException(access_response)
 
-    if check_if_enrolled:
-        # If the user is not enrolled, redirect them to the about page
-        if not CourseEnrollment.is_enrolled(user, course.id):
-            raise CourseAccessRedirect(reverse('about_course', args=[unicode(course.id)]))
+    # if check_if_enrolled:
+    #     # If the user is not enrolled, redirect them to the about page
+    #     if not CourseEnrollment.is_enrolled(user, course.id):
+    #         raise CourseAccessRedirect(reverse('about_course', args=[unicode(course.id)]))
```

Comment:
this code block will redirect the user into the `about` page if the user is not enrolled, and the about page will redirect the user to here and so on....


This what will happen after I commented out this block, the user will have an alert saying you should be enrolled 

<img width="1440" alt="Screen Shot 2020-11-16 at 12 29 17 PM" src="https://user-images.githubusercontent.com/38977667/99242048-5cb24080-2807-11eb-95e5-d99b65a0a680.png">

Reply:
I've disabled redirection in another place, this will:
1- avoid breaking tests
2- avoid disabling redirection completely